tests/lib/rules/prefer-find-by.test.ts

```diff
@@ -0,0 +1,80 @@
+import { InvalidTestCase } from '@typescript-eslint/experimental-utils/dist/ts-eslint'
+import { createRuleTester } from '../test-utils';
+import { ASYNC_QUERIES_COMBINATIONS, SYNC_QUERIES_COMBINATIONS } from '../../../lib/utils';
+import rule, { WAIT_METHODS, RULE_NAME } from '../../../lib/rules/prefer-find-by';
+
+const ruleTester = createRuleTester({
+  ecmaFeatures: {
+    jsx: true,
+  },
+});
+
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
+    ...ASYNC_QUERIES_COMBINATIONS.map((queryMethod: string) => ({
+      code: `const submitButton = await ${queryMethod}('foo')`
+    })),
+    ...ASYNC_QUERIES_COMBINATIONS.map((queryMethod: string) => ({
+      code: `const submitButton = await screen.${queryMethod}('foo')`
+    })),
+    ...SYNC_QUERIES_COMBINATIONS.map((queryMethod: string) => ({
+      code: `await waitForElementToBeRemoved(() => ${queryMethod}(baz))`
+    })),
+    ...SYNC_QUERIES_COMBINATIONS.map((queryMethod: string) => ({
+      code: `await waitFor(function() {
+        return ${queryMethod}('baz', { name: 'foo' })
+      })`
+    })),
+    {
+      code: `await waitFor(() => myCustomFunction())`
+    },
+    {
+      code: `await waitFor(customFunctionReference)`
+    },
+    {
+      code: `await waitForElementToBeRemoved(document.querySelector('foo'))`
+    },
+    ...SYNC_QUERIES_COMBINATIONS.map((queryMethod: string) => ({
+      code: `
+        await waitFor(() => {
+          foo()
+          return ${queryMethod}()
+        })
+      `
+    })),
```

Comment:
Could you add valid tests for the example requested in the doc? A valid `waitFor` returning expect with getBy inside.

Reply:
done